torchgen/dest/register_dispatch_key.py

```diff
@@ -409,6 +416,77 @@ def generate_defn(cpp_sig: CppSignature) -> str:
                 metadata = self.backend_index.get_kernel(f)
                 if metadata is None:
                     return None
+
+                # define the struct for dispatch-less composite kernels.
+                impl_struct = ""
+                if self.backend_index.should_gen_dispatchless_composite(f):
+                    # define the native function as a wrapper static method, redirecting the call to
+                    # the appropriate function (e.g. on CPU, on CUDA, or as a CompositeExplicitAutograd).
+                    def dispatchless_function_defn(
+                        f: NativeFunction, g: Optional[NativeFunctionsGroup]
+                    ) -> str:
+                        dispatch_key = self.backend_index.dispatch_key
+
+                        # the declaration signature is a static method (not a Tensor method).
+                        decl_sig = CppSignatureGroup.from_native_function(
+                            f, method=False, fallback_binding=False
+                        ).signature
+
+                        # decide 2 things:
+                        #     1. where is the redirect function? (i.e. its namespace)
+                        #     2. is it a Tensor method? (some operations are only available as methods)
+                        if self.backend_index.has_registered_kernel(f, g):
+                            prefix = f"at::{dispatch_key.lower()}::"
+                            is_method = False
+                        else:
```

Comment:
This else branch is very unsatisfactory

Reply:
Hmm. I did it so that I wouldn't have to repeat the definition of `is_method`. But I agree that repeating is better, here.